ecosystem/node-checker/src/evaluators/direct/node_identity.rs

```diff
@@ -0,0 +1,153 @@
+                e
+            )
+        })?;
+
+    let chain_id_raw: u8 = data
+        .get("chain_id")
+        .ok_or_else(|| format_err!("Failed to get chain_id from node identity"))?
+        .as_u64()
+        .ok_or_else(|| anyhow!("Failed to read chain ID from node identity as u8"))?
+        as u8;
+    let chain_id = ChainId::new(chain_id_raw);
+
+    let role_type_raw = data
+        .get("node_role")
+        .ok_or_else(|| format_err!("Failed to get node_role from node identity"))?
+        .as_str()
+        .ok_or_else(|| anyhow!("Failed to read node_role from node identity as str"))?;
+    let role_type = RoleType::from_str(role_type_raw)
+        .map_err(|e| format_err!("Failed to parse node_role {}", e))?;
+
+    Ok((chain_id, role_type))
+}
+
+#[derive(Debug, ThisError)]
+pub enum NodeIdentityEvaluatorError {
+    #[error("Failed to get node identity from node: {0}")]
+    GetNodeIdentityError(anyhow::Error),
+}
+
+// TODO: Consider taking chain_id and role_type here instead.
+#[derive(Clone, Debug, Deserialize, Parser, PoemObject, Serialize)]
+pub struct NodeIdentityEvaluatorArgs {}
+
+#[allow(dead_code)]
+#[derive(Debug)]
+pub struct NodeIdentityEvaluator {
+    args: NodeIdentityEvaluatorArgs,
+}
+
+impl NodeIdentityEvaluator {
+    pub fn new(args: NodeIdentityEvaluatorArgs) -> Self {
+        Self { args }
+    }
+
+    fn build_evaluation_result(
+        &self,
+        headline: String,
+        score: u8,
+        explanation: String,
+    ) -> EvaluationResult {
+        EvaluationResult {
+            headline,
+            score,
+            explanation,
+            category: CATEGORY.to_string(),
+            evaluator_name: Self::get_name(),
+            links: vec![],
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl Evaluator for NodeIdentityEvaluator {
+    type Input = DirectEvaluatorInput;
+    type Error = NodeIdentityEvaluatorError;
+
+    /// Assert that the node identity (role type and chain ID) of the two
+    /// nodes match.
+    async fn evaluate(&self, input: &Self::Input) -> Result<Vec<EvaluationResult>, Self::Error> {
+        let (target_chain_id, target_role_type) = get_node_identity(&input.target_node_address)
+            .await
+            .map_err(NodeIdentityEvaluatorError::GetNodeIdentityError)?;
+
+        let evaluation_result = if (input.baseline_node_information.chain_id == target_chain_id)
+            && (input.baseline_node_information.role_type == target_role_type)
+        {
+            self.build_evaluation_result(
+                "Chain ID and role type reported by baseline and target match".to_string(),
+                100,
+                format!(
+                    "The node under investigation reported the same chain ID \"{}\" \
+                and role type \"{}\" as is reported by the baseline node",
+                    target_chain_id, target_role_type
+                ),
+            )
+        } else {
+            self.build_evaluation_result(
```

Comment:
nit: should we be more explicit and check each independently? Just thinking about the users and also any UIs built on top of this 🤔

Reply:
Yeah sure seems like a good idea.